allennlp/tests/training/learning_rate_schedulers_test.py

```diff
@@ -321,16 +321,22 @@ def test_schedules(self):
         for params, lr_checks in slanted_triangular_cases:
             optimizer = self._get_optimizer()
             params["type"] = "slanted_triangular"
-            scheduler = LearningRateScheduler.from_params(optimizer, Params(params)).lr_scheduler
+            scheduler = LearningRateScheduler.from_params(optimizer, Params(params))
             lrs = []
 
+            batch_num_total = 0
             for epoch in range(params["num_epochs"]):
                 for _ in range(params["num_steps_per_epoch"]):
-                    scheduler.step_batch()
-                    lrs.append([param_group["lr"] for param_group in optimizer.param_groups])
+                    batch_num_total += 1
+                    # allennlp trainer calls step_batch after updating parameters
+                    # so collect lr at time of parameter update
+                    lrs.append([param_group["lr"] * float(param_group['params'][0].requires_grad)
```

Comment:
This also checks that the `requires_grad` flag is set as expected by setting the effective learning rate to 0.0 for parameters that are frozen.